### PR TITLE
chore: openapi editorial

### PR DESCRIPTION
**Pull request by mfw78**
This PR:

1. Contains editorial changes to the *descriptions* and *summary* only for the `OrderBook` OpenAPI specification. Notably all descriptions / summaries:
  a. Adhere to UK English.
  b. Single-word descriptions / summaries are not punctuated. Otherwise, they are punctuated.
  c. Standardises hyphenation.
  d. Makes use of Markdown within the description with respect to cautions and notes.
  
**Why this PR?**

As part of expanding `cow-sdk` for documentation and usability for `ComposableCoW`, `openapi.yml` was reviewed to for editorial oversights / typos / grammer / formatting. This is due to `openapi.yml` being imported into `cow-sdk` for generated bindings.

### Test Plan

There is no change to code coverage, therefore testing is not required.

### CODE REVIEW

**Comment by mfw78** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nlordell** [comment]
> There is no change to code coverage, therefore testing is not required.

The OpenAPI YAML gets validated in CI as well, so the changes should not have invalidated the spec.

File: crates/orderbook/openapi.yml [hunk 1]
```diff
@@ -91,56 +92,58 @@ paths:
...
       deprecated: true
-      summary: Cancels order by marking it invalid with a timestamp.
+      summary: Cancel an order by marking it invalid with a timestamp.
       description: |
         The successful deletion might not prevent solvers from settling the order.
-        Authentication must be provided by providing an EIP-712 signature of an
-        "OrderCacellation(bytes orderUids)" message.
+        Authentication must be provided by providing an 
+        [EIP-712](https://eips.ethereum.org/EIPS/eip-712) signature of an
+        `OrderCancellation(bytes orderUids)` message.
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:103]:
```suggestion
        `OrderCancellation(bytes orderUid)` message.
```

Just checked, and it looks like this was a typo in the original OpenAPI spec:

https://github.com/cowprotocol/services/blob/7c2da6d7a2b0e68c07610d277c415cbbb26a5085/crates/model/src/order.rs#L536-L539

File: crates/orderbook/openapi.yml [hunk 2]
```diff
@@ -48,30 +48,31 @@ paths:
...
+        description: Signed `OrderCancellations`.
         required: true
         content:
           application/json:
             schema:
               $ref: "#/components/schemas/OrderCancellations"
       responses:
         200:
-          description: Orders deleted
+          description: Order(s) are deleted.
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:67]:
```suggestion
          description: Order(s) are cancelled.
```

File: crates/orderbook/openapi.yml [hunk 3]
```diff
@@ -91,56 +92,58 @@ paths:
...
+        description: Signed `OrderCancellation`
         required: true
         content:
           application/json:
             schema:
               $ref: "#/components/schemas/OrderCancellation"
       responses:
         200:
-          description: Order deleted
+          description: Order deleted.
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:119]:
```suggestion
          description: Order cancelled.
```

File: crates/orderbook/openapi.yml [hunk 4]
```diff
@@ -229,9 +242,9 @@ paths:
     get:
       summary: Get orders of one user paginated.
       description: |
-        The orders are ordered by their creation date descending (newest orders first).
-        To enumerate all orders start with offset 0 and keep increasing the offset by the total
-        number of returned results. When a response contains less than the limit the last page has
+        The orders are sorted by their creation date descending (newest orders first).
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:245]: The pun was so good though 😛

File: crates/orderbook/openapi.yml [hunk 5]
```diff
@@ -345,39 +357,40 @@ paths:
...
-          description: version info
+          description: Version
           content:
             application/json:
               schema:
                 $ref: "#/components/schemas/VersionResponse"
   /api/v1/app_data/{app_data_hash}:
     get:
-      summary: Retrieve full app data from contract app data hash.
+      summary: Get the full appData from contract appDataHash.
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:393]: Should you `\`` for consistency?

```suggestion
      summary: Get the full `appData` from contract `appDataHash`.
```

File: crates/orderbook/openapi.yml [hunk 6]
```diff
@@ -625,33 +643,46 @@ components:
...
           description: |
             Orders can optionally include a quote ID. This way the order can be linked to a quote
-            and enable providing more metadata when analyzing order slippage.
+            and enable providing more metadata when analysing order slippage.
           type: integer
           nullable: true
         appData:
           description: |
-            This field comes in two forms for backward compatibility. The first form (app data hash) will eventually stop being accepted.
+            This field comes in two forms for backward compatibility. The first form (app data hash)
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:651]:
```suggestion
            This field comes in two forms for backward compatibility. The first form (`appDataHash`) 
```

File: crates/orderbook/openapi.yml [hunk 7]
```diff
@@ -843,7 +881,8 @@ components:
         - signingScheme
     OrderCancellation:
       description: |
-        EIP-712 signature of struct OrderCancellation { orderUid: bytes } from the order's owner
+        [EIP-712](https://eips.ethereum.org/EIPS/eip-712) signature of struct 
+        OrderCancellation { orderUid: bytes } from the order's owner.
```
Comment (nlordell) [inline crates/orderbook/openapi.yml:885]: Would suggest keeping the EIP-712 notation consistent:

```suggestion
        `OrderCancellation(bytes orderUid)` from the order's owner.
```

**Review by nlordell — APPROVED** [review]
Awesome! Left some nits/suggestions (mostly from things that were wrong before as well and went unnoticed that I saw now when reviewing the changes).

File: crates/orderbook/openapi.yml (hunk 5 above)
Comment (mfw78) [reply crates/orderbook/openapi.yml:393]: Nice catch, by this time it started to all blur together 🫨

**Review by vkgnosis — APPROVED** [review]

**Review by MartinquaXD — APPROVED** [review]

**Comment by nlordell** [comment]
Looks like this needs a rebase/merge from `main`, then it's ready to go.

**Comment by vkgnosis** [comment]
Slightly annoying because we have to approve the workflow to run to merge but by then the branch might me out of date already. I will update the branch to master for you and merge it to make sure it gets into main now.

**Comment by vkgnosis** [comment]
Ah I think I can't because you have turned off "allow repository maintainers to make changes to my PR".

**Comment by mfw78** [comment]
> Ah I think I can't because you have turned off "allow repository maintainers to make changes to my PR".

@vkgnosis Sorry for the "noobness" - despite much Googling, I've come up short of exactly where to allow this.

**Comment by vkgnosis** [comment]
No worries. I don't know where exactly it is. I just remember seeing it when I made PRs before.

**Pull request merged**